momepy/preprocessing.py

```diff
@@ -1101,3 +1105,464 @@ def roundabout_simplification(
+    )
+
+    # Determine what edges need to be simplified (either between diff.
+    #  clusters or self-loops in a cluster):
+    edges_tosimplify_gdf = edges_gdf.query(
+        f"origin_cluster != destination_cluster or (('{edge_to_col}' == '{edge_from_col}') and origin_cluster >= 0)"
+    )
+
+    # Determine the new point geometries (when exists):
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_origin_pt=edges_tosimplify_gdf.origin_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+    edges_tosimplify_gdf = edges_tosimplify_gdf.assign(
+        new_destination_pt=edges_tosimplify_gdf.destination_cluster.map(
+            cluster_gdf.geometry, None
+        )
+    )
+
+    # Determine the new geometry according to the simplification method:
+    if method == "extend":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _extension_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "euclidean":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _euclidean_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "spider":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    else:
+        warnings.warn(
+            f"Simplification method '{method}' not recognized. Using spider-web simplification."
+        )
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+
+    # Rename and update the columns:
+    cols_rename = {
+        edge_from_col: "original_from",
+        edge_to_col: "original_to",
+        "origin_cluster": edge_from_col,
+        "destination_cluster": edge_to_col,
+        "geometry": "original_geometry",
+        "new_geometry": "geometry",
+    }
+    cols_drop = ["new_origin_pt", "new_destination_pt"]
+    new_edges_gdf = edges_simplified_gdf.rename(cols_rename, axis=1).drop(
```

Comment:
I am not entirely sure how geopandas handles this specific renaming of geometry columns but I would rather throw in `set_geometry('geometry')` just to be sure the right column is set as active.